packages/financial-templates-lib/test/clients/OptimisticOracleClient.js

```diff
@@ -126,203 +127,418 @@ describe("OptimisticOracleClient.js", function () {
-      .disputePrice(optimisticRequester.options.address, identifier, requestTime, "0x")
-      .send({ from: disputer });
-    result = client.getSettleableDisputes(disputer);
-    objectsInArrayInclude(result, []);
-
-    // Resolve the dispute and check that the client detects the new state:
-    await pushPrice(correctPrice);
-    await client.update();
-    // Note: `getSettleableDisputes` only returns proposals where the `disputer` is involved
-    result = client.getSettleableDisputes(rando);
-    objectsInArrayInclude(result, []);
-    result = client.getSettleableDisputes(disputer);
-    objectsInArrayInclude(result, [
-      {
-        requester: optimisticRequester.options.address,
-        proposer: proposer,
-        disputer: disputer,
-        identifier: hexToUtf8(identifier),
-        ancillaryData: "0x",
-        timestamp: requestTime.toString(),
-      },
-    ]);
-
-    // Settle the dispute and make sure that the client no longer sees it as settleable:
-    await optimisticOracle.methods
-      .settle(optimisticRequester.options.address, identifier, requestTime, "0x")
-      .send({ from: accounts[0] });
-    await client.update();
-    result = client.getSettleableDisputes(disputer);
-    objectsInArrayInclude(result, []);
+    it("Lookback window enforced", async function () {
+      // Create a new client with a shorter lookback equal to approximately
+      // the amount of seconds that it takes 1 block to get mined
+      let clientShortLookback = new OptimisticOracleClient(
+        dummyLogger,
+        OptimisticOracle.abi,
+        MockOracle.abi,
+        web3,
+        optimisticOracle.options.address,
+        mockOracle.options.address,
+        13
+      );
+
+      // Request a price and check that the longer lookback client currently sees it
+      await optimisticRequester.methods
+        .requestPrice(identifier, requestTime, "0x", collateral.options.address, 0)
+        .send({ from: accounts[0] });
+      await client.update();
+      let result = client.getUnproposedPriceRequests();
+      objectsInArrayInclude(result, [
+        {
+          requester: optimisticRequester.options.address,
+          identifier: hexToUtf8(identifier),
+          ancillaryData: "0x",
+          timestamp: requestTime.toString(),
+          currency: collateral.options.address,
+          reward: "0",
+          finalFee,
+        },
+      ]);
+
+      // Mine two blocks to move past the lookback window, and make sure the shorter lookback client
+      // ignores the price request.
+      const currentTime = Number((await web3.eth.getBlock("latest")).timestamp);
+      await advanceBlockAndSetTime(web3, currentTime + 1);
+      await advanceBlockAndSetTime(web3, currentTime + 2);
+
+      await clientShortLookback.update();
+      result = clientShortLookback.getUnproposedPriceRequests();
+      objectsInArrayInclude(result, []);
+    });
   });
+  describe("Skinny OptimisticOracle contract", function () {
```

Comment:
These tests are new and follow the exact same flow as in the traditional OptimisticOracle tests